exercises/triangle/examples/src/Triangle.purs

```diff
@@ -0,0 +1,41 @@
+module Triangle
+  ( Triangle(Equilateral, Isosceles, Scalene)
+  , triangleKind
+  ) where
+
+import Prelude
+import Data.Set as Set
+import Data.Either (Either(Left, Right))
+import Data.Generic (class Generic, gShow, gEq)
+
+
+data Triangle
+  = Equilateral
+  | Isosceles
+  | Scalene
+
+derive instance genericTriangle :: Generic Triangle
+
+instance eqTriangle :: Eq Triangle where
+  eq = gEq
+
+instance showTriangle :: Show Triangle where
+  show = gShow
+
+
+type ErrorMsg = String
+
+
+triangleKind :: Int -> Int -> Int -> Either Triangle ErrorMsg
```

Comment:
Thanks! In Rust there's `Ok` and `Err`, which I find easier to read than `Left` and `Right`. Is there anything like this?